tests/src/Kernel/TypedData/DataFetcherTest.php

```diff
@@ -70,4 +82,85 @@ public function testFetchingByBasicPropertyPath() {
     );
   }
 
-}
\ No newline at end of file
+  /**
+   * @cover fetchBySubPaths
+   */
+  public function testFetchingByBasicSubPath() {
+    $this->assertEquals(
+      $this->node->title->value,
+      $this->typedDataManager->getDataFetcher()
+      ->fetchBySubPaths($this->node->getTypedData(), array('title', '0', 'value'))
+      ->getValue()
+    );
+  }
+
+  /**
+   * @cover fetchByPropertyPath
+   */
+  public function testFetchingEntityReference() {
+    $user = $this->entityTypeManager->getStorage('user')
+      ->create([
+        'name' => 'test',
+        'type' => 'user',
+      ]);
+    $this->node->uid->target_id = $user->id();
+
+    $fetched_user = $this->typedDataManager->getDataFetcher()
+      ->fetchByPropertyPath($this->node->getTypedData(), 'uid.entity.value')
+      ->getValue();
+    $this->assertEquals(TRUE, $fetched_user instanceof EntityAdapter);
+  }
+
+  /**
+   * @cover fetchByPropertyPath
+   */
+  public function testFetchingEntityReferenceAtPosition0() {
+    $user = $this->entityTypeManager->getStorage('user')
+      ->create([
+        'name' => 'test',
+        'type' => 'user',
+      ]);
+    $this->node->uid->target_id = $user->id();
+
+    $fetched_user = $this->typedDataManager->getDataFetcher()
+      ->fetchByPropertyPath($this->node->getTypedData(), 'uid.0.entity.value')
+      ->getValue();
+    $this->assertEquals(TRUE, $fetched_user instanceof EntityAdapter);
+  }
+
+  /**
+   * @cover fetchByPropertyPath
+   */
+  public function testFetchingEntityReferenceAtPosition1() {
+    $user1 = $this->entityTypeManager->getStorage('user')
+      ->create([
+        'name' => 'test1',
+        'type' => 'user',
+      ]);
+    $users[]['target_id'] = $user1->id();
+    $user2 = $this->entityTypeManager->getStorage('user')
+      ->create([
+        'name' => 'test2',
+        'type' => 'user',
+      ]);
+    $users[]['target_id'] = $user2->id();
+    $this->node->uid->setValue($users);
+
+    $fetched_user = $this->typedDataManager->getDataFetcher()
+      ->fetchByPropertyPath($this->node->getTypedData(), 'uid.1.entity.value')
+      ->getValue();
+    $this->assertEquals(TRUE, $fetched_user instanceof EntityAdapter);
+  }
+
+  /**
+   * @cover fetchByPropertyPath
+   */
+  public function testFetchingNonExistingEntityReference() {
+    $this->setExpectedException('Drupal\Core\TypedData\Exception\MissingDataException');
```

Comment:
usual there is some expectedException annotation, look at core examples or other rules examples